src/python/dxpy/utils/executable_unbuilder.py

```diff
@@ -140,48 +174,47 @@ def make_cluster_bootstrap_script_file(region, entry_point, code, suffix):
-            asset_record_id = file_handle.get_properties().get("AssetBundle")
-            asset_record = None
-            if asset_record_id:
-                asset_record = dxpy.DXRecord(asset_record_id)
-                if asset_record:
-                    try:
-                        asset_json = {"name": asset_record.describe().get("name"),
-                                              "project": asset_record.get_proj_id(),
-                                              "folder": asset_record.describe().get("folder"),
-                                              "version": asset_record.describe(fields={"properties": True}
-                                                                               )["properties"]["version"]
-                                              }
-                        if dep.get("stages"):
-                            asset_json["stages"] = dep["stages"]
-                        asset_depends.append(asset_json)
-                        deps_to_remove.append(dep)
-                    except DXError:
-                        print("Describe failed on the assetDepends record object with ID - " +
-                              asset_record_id + "\n", file=sys.stderr)
-                        pass
-            else:
-                break
-    # Reversing the order of the asset_depends[] so that original order is maintained
-    asset_depends.reverse()
-    # resources/ directory
+    download_completed = omit_resources
+    deps_downloaded = set()
+    deps_assets = set()
     created_resources_directory = False
-    if not omit_resources:
-        for dep in info["runSpec"]["bundledDepends"]:
-            if dep in deps_to_remove:
-                continue
-            handler = get_handler(dep["id"])
-            if isinstance(handler, dxpy.DXFile):
+    # Download resources
+    for region in enabled_regions:
+        # check if downloading has already completed
+        # already True if omits resources 
+        if download_completed:
+            break
+        
+        print("Trying to download resources from region {}...".format(region))
+
+        for dep in info["runSpec"]["bundledDependsByRegion"][region]:
+            try: 
+                # if retrying another region, skip the ones identified as assets
+                if dep.get("name") in deps_assets:
+                    continue
+
+                file_handle = get_handler(dep["id"])
+                handler_id = file_handle.get_id()
+                # if dep is not a file (record etc.), check the next dep
+                if not isinstance(file_handle, dxpy.DXFile):
+                    continue
+                
+                # check if the file is an asset dependency
+                # if so, skip downloading
+                # if failed to describe it, will try the next region
+                if file_handle.get_properties().get("AssetBundle"):
+                    deps_assets.add(dep.get("name"))
+                    continue   
```

Comment:
downloading is skipped if:

- it is not a file
- it is an asset
- it has been identified as an asset in the past but failed loops